.travis.yml

```diff
@@ -50,7 +50,7 @@ script:
     fi
   - mkdir build
   - cd build
-  - ../generate_makefile.bash --compiler=$CXX --with-$THREADING --with-options=compiler_warnings ${GENERATE_OPTS}
+  - ../generate_makefile.bash --compiler=$CXX --with-$THREADING --with-options=compiler_warnings --cxxflags=-Werror ${GENERATE_OPTS}
```

Comment:
You might want to pass flags to print more warning message.  Refer to this
https://github.com/kokkos/kokkos/blob/5d6e7fb38e96aec88d2c514e1f9be1cf2b549b57/scripts/testing_scripts/test_all_sandia#L65-L69

Reply:
With
```
../generate_makefile.bash --compiler=clang++ --with-options=compiler_warnings --cxxflags=-Werror
```
a typical compiler call looks like
```
 /opt/compiler/llvm-6.0.0/bin/clang++  -I./ -I/mnt/data/darndt/Sources/kokkos/build/install/include -I/mnt/data/darndt/Sources/kokkos/build/install/include -I/mnt/data/darndt/Sources/kokkos/build/install/include -I/mnt/data/   darndt/Sources/kokkos/build/install/include/eti -Wall -Wshadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized --std=c++11 --gcc-toolchain=/export/home/darndt/.linuxbrew -Werror  -c /mnt/data/darndt/Sources/kokkos/     example/tutorial/Advanced_Views/03_subviews/subviews.cpp -o subviews.o
```
In particular, `--with-options=compiler_warnings` enables all these warnings already.

Reply:
Didn't know about the `compiler_warnings` option.  Thanks :)